networkx/drawing/layout.py

```diff
@@ -485,9 +492,20 @@ def fruchterman_reingold_layout(
             # We must adjust k by domain size for layouts not near 1x1
             nnodes, _ = A.shape
             k = dom_size / np.sqrt(nnodes)
-        pos = _fruchterman_reingold(
-            A, k, pos_arr, fixed, iterations, threshold, dim, seed
-        )
+        # In the case where no node_weight is given the standard fruchterman_reingold function can be used
+        if node_weight is None:
+            pos = _fruchterman_reingold(
+                A, k, pos_arr, fixed, iterations, threshold, dim, seed
+            )
+        else:  # Otherwise, we must use our extended function
+            mass_array = np.asarray(
+                [G.nodes[node][node_weight] for node in G], dtype="f"
+            )
+            # Generate the matrix $B$ by taking the outer product of mass_array with itself (so $B_{ij} = m_i m_j$)
+            B = np.tensordot(mass_array, mass_array, axes=0)
```

Comment:
```suggestion
            B = np.outer(mass_array, mass_array)
```
`tensordot` might be overkill - do we ever expect `node_weight` to be non-scalar?